iphone/Maps/Core/Location/MWMLocationManager.mm

```diff
@@ -469,43 +488,25 @@ - (void)setStarted:(BOOL)started
   }
 }
 
-- (BOOL)start
+- (void)requestPermissionIfNeededAndStartUpdatingLocationFor:(CLLocationManager *)manager
 {
-  MWMVoidBlock doStart = ^{
-    LOG(LINFO, ("startUpdatingLocation"));
-
-    CLLocationManager * locationManager = self.locationManager;
-    if ([locationManager respondsToSelector:@selector(requestWhenInUseAuthorization)])
-      [locationManager requestWhenInUseAuthorization];
-
-    [locationManager startUpdatingLocation];
-
-    setPermissionRequested();
-
-    if ([CLLocationManager headingAvailable])
-      [locationManager startUpdatingHeading];
-  };
+  LOG(LINFO, ("startUpdatingLocation"));
+  
+  if ([manager respondsToSelector:@selector(requestWhenInUseAuthorization)])
+    [manager requestWhenInUseAuthorization];
+  
+  [manager startUpdatingLocation];
```

Comment:
Should this call be done only when a user approves the permissions?

Does it make sense to split permission requests from starting location updates, and start location updates only when permission was granted (or previously granted, and did not change since then)?

Is there a difference for iOS 12 and 13 that do not support `didChangeAuthorizationStatus` ?

Reply:
1. if user press `Allow Once` or `Dont Allow` the permission alert will not be shown when this method calling.
So it is safe to call multiple times without afraid of seen system alert.
2. Yes! Good point! I will separate them. Becaulse app can handle location updates without permission.
3. I am using the old version of the delegate method which is supported by ios 12. One time we will replace it with the newer version
https://developer.apple.com/documentation/corelocation/cllocationmanagerdelegate/1423701-locationmanager?language=objc